programs/lighthouse/src/utils.rs

```diff
@@ -18,9 +18,23 @@ use solana_program::{
 
 pub type Result<T> = std::result::Result<T, ProgramError>;
 
-pub fn unpack_coption_key(src: &[u8]) -> Result<Option<&Pubkey>> {
-    let tag = &src[0..4];
-    let body = &src[4..36];
+pub fn checked_get_slice(data: &[u8], offset: usize, length: usize) -> Result<&[u8]> {
+    let end = offset + length;
+    data.get(offset..end)
+        .ok_or_else(|| LighthouseError::oob_err(offset..end))
+}
+
+pub fn unpack_coption_key(src: &[u8], offset: usize) -> Result<Option<&Pubkey>> {
```

Comment:
I feel like it'd be semantically cleaner to deserialize a `COption` with `try_from_slice` and then manually cast that to an `Option`, but it doesn't matter too much